File: library-test/src/androidTest/java/com/uphyca/creditcardedittext/subject/CreditCardDateEditTextSubject.java
package com.uphyca.creditcardedittext.subject;

import com.google.common.truth.FailureMetadata;
import com.google.common.truth.Subject;
import com.uphyca.creditcardedittext.CreditCardDate;
import com.uphyca.creditcardedittext.CreditCardDateEditText;

import static com.google.common.truth.Truth.assertAbout;

/**
 * Subject for {@link CreditCardDateEditText} instances.
 */
public class CreditCardDateEditTextSubject extends Subject {

    public static CreditCardDateEditTextSubject assertThat(CreditCardDateEditText editText) {
        return assertAbout(editTexts()).that(editText);
    }

    private static Subject.Factory<CreditCardDateEditTextSubject, CreditCardDateEditText> editTexts() {
        return CreditCardDateEditTextSubject::new;
    }

    private final CreditCardDateEditText actual;

    private CreditCardDateEditTextSubject(FailureMetadata failureMetadata, CreditCardDateEditText actual) {
        super(failureMetadata, actual);
        this.actual = actual;
    }

    public final void hasDate(CreditCardDate date) {
        check("getNumber()")
                .that(actual.getDate())
                .isEqualTo(date);
    }

    public final void hasTextString(String number) {
        check("getNumber()")
                .that(actual.getText().toString())
                .isEqualTo(number);
    }
}

File: library-test/src/androidTest/java/com/uphyca/creditcardedittext/subject/CreditCardDateEditTextSubject.kt
package com.uphyca.creditcardedittext.subject

import com.google.common.truth.FailureMetadata
import com.google.common.truth.Subject
import com.google.common.truth.Subject.Factory
import com.google.common.truth.Truth.assertAbout
import com.uphyca.creditcardedittext.CreditCardDate
import com.uphyca.creditcardedittext.CreditCardDateEditText

/**
 * Subject for [CreditCardDateEditText] instances.
 */
class CreditCardDateEditTextSubject private constructor(
    failureMetadata: FailureMetadata,
    private val actual: CreditCardDateEditText
) : Subject(failureMetadata, actual) {

    fun hasDate(date: CreditCardDate?) {
        check("getNumber()")
            .that(actual.date)
            .isEqualTo(date)
    }

    fun hasTextString(number: String?) {
        check("getNumber()")
            .that(actual.text.toString())
            .isEqualTo(number)
    }

    companion object {

        @JvmStatic
        fun assertThat(editText: CreditCardDateEditText?): CreditCardDateEditTextSubject {
            return assertAbout(editTexts()).that(editText)
        }

        private fun editTexts(): Factory<CreditCardDateEditTextSubject, CreditCardDateEditText> {
            return Factory { failureMetadata, actual ->
                CreditCardDateEditTextSubject(failureMetadata, actual)
            }
        }
    }
}
